### PR TITLE
feat: propagate_tags tactic

### PR DESCRIPTION
**Pull request by Beanway144**
Adds `propagate_tags`

### CODE REVIEW

File: Mathlib/Tactic/PropagateTags.lean [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+Author: Ian Benway.
+-/
+
+import Lean
+namespace Mathlib.Tactic
+open Lean Elab.Tactic Meta
+
+/-- `guard_tags t` tests that `t` is the tag of the main goal. -/
+elab (name := guardTags) "guard_tags" t:(ppSpace ident) : tactic => do
+  guard ((← getMainTag) == t.getId)
```
Comment (digama0) [inline Mathlib/Tactic/PropagateTags.lean:13]:
```suggestion
/-- `guard_tag t` tests that `t` is the tag of the main goal. -/
elab (name := guardTag) "guard_tag" t:(ppSpace ident) : tactic => do
  guard ((← getMainTag) == t.getId)
```

File: Mathlib/Tactic/PropagateTags.lean [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+/--
+`propagate_tags tac` carries the tag of the main goal to the main goal of the result of doing `tac`.
+-/
+elab "propagate_tags " tac:tacticSeq : tactic => do
+  withMainContext do
```
Comment (digama0) [inline Mathlib/Tactic/PropagateTags.lean:19]:
```suggestion
elab "propagate_tags " tac:tacticSeq : tactic => withMainContext do
```

**Comment by semorrison** [comment]
@Beanway144, sorry this languished on the PR queue for a while. Would you be able to merge the conflict and adopt Mario's suggestions?

**Comment by jcommelin** [comment]
This pull request has not seen any activity for quite a while, and it does not pass all CI checks. For these reasons, I am closing it now.

If you are still interested in getting this PR merged, please re-open the PR, update the branch, and make sure that all CI checks pass, Please label the PR with `awaiting-review` when ready.

**Pull request closed without merging**